### DISCUSSION TITLE
2FA only for some users

### DISCUSSION THREAD

**Discussion by michalsn** [codeigniter4/shield#940] [Q&A] [answered]
Is there an option to set 2FA when logging in only for certain users? Preferably for those who have enabled it themselves?

**Comment by datamweb** [comment, marked as answer]
@michalsn Your answer is NO.
I was going to add something similar to this feature in PR #324, but it didn't integrate.
Currently, it seems that you have to write [defining-new-actions](https://codeigniter4.github.io/shield/references/authentication/auth_actions/#defining-new-actions).

**Reply by michalsn** [reply]
Ok, thanks! I will look into this.

**Reply by datamweb** [reply]
@michalsn see this [link](https://github.com/codeigniter4/shield/discussions/525#discussioncomment-4192164).

**Reply by michalsn** [reply]
Perfect. Thank you.
